internal/types/document_test.go

```diff
@@ -161,4 +162,57 @@ func TestDocument(t *testing.T) {
 			})
 		}
 	})
+
+	t.Run("Next", func(t *testing.T) {
+		t.Parallel()
+
+		type fields struct {
+			v any
+			k string
+		}
+
+		for name, tc := range map[string]struct {
+			document *Document
+			expected []fields
+		}{
+			"empty": {
+				document: must.NotFail(NewDocument()),
+				expected: []fields{},
+			},
+			"one": {
+				document: must.NotFail(NewDocument("foo", "bar")),
+				expected: []fields{{k: "foo", v: "bar"}},
+			},
+			"two": {
+				document: must.NotFail(NewDocument("foo", "bar", "baz", "qux")),
+				expected: []fields{{k: "foo", v: "bar"}, {k: "baz", v: "qux"}},
+			},
+			"three": {
+				document: must.NotFail(NewDocument("foo", "bar", "baz", "qux", "quux", "quuz")),
+				expected: []fields{{k: "foo", v: "bar"}, {k: "baz", v: "qux"}, {k: "quux", v: "quuz"}},
```

Comment:
I'd suggest having at lease one duplicate key, as this is one of the reasons why we want to have iterator, something like:

```suggestion
				document: must.NotFail(NewDocument("foo", "bar", "baz", "qux", "foo", "quuz")),
				expected: []fields{{k: "foo", v: "bar"}, {k: "baz", v: "qux"}, {k: "foo", v: "quuz"}},
```